lib/class-perfecty-push-lib-push-server.php

```diff
@@ -84,7 +84,7 @@ public static function create_vapid_keys() {
 	 * @return int $notification_id if success, false otherwise
 	 * @throws Exception
 	 */
-	public static function schedule_broadcast_async( $payload ) {
+	public static function schedule_broadcast_async( $payload, $timeoffset = 0 ) {
```

Comment:
`schedule_broadcast_async` will be used in the integrations with other plugins that call us using PHP.

 Is it possible to make the timeoffset calculation in the PHP side so that it also works with external plugins. `schedule_broadcast_async` could receive a `DateTime` parameter and we can calculate the offset, like:

```php
public static function schedule_broadcast_async( $payload, $datetime = null ) {
  $timeoffset = $datetime == null ? 0 : calculate_offset($datetime);
```

`calculate_offset()` will be the equivalent to the JS code:

```js
var scheduledTimeEpoch = Math.round(scheduledTime.getTime() / 1000);
var currentTimeEpoch = Math.round(Date.now() / 1000);
var timeOffset = scheduledTimeEpoch - currentTimeEpoch;
```

Reply:
I will work on this, but it is the real problem: we could have users (browsers) in a timezone different from the server time zone. This is why I calculated offset on JS. When a user schedules a notifications he watches at its clock (browser time) and it could be hours off from server time.
Yes, maybe the best thing, is to calculate offset in js then parse the offset in PHP side to convert it in a date time in server timezone (this will be the scheduled date time recorded for the cron hook), and then convert it back to an offset inside the schedule_broadcast_async...
Give me some time to think about this

Reply:
I have made some changes. Now schedule_broadcast_async( $payload, $scheduled_time = null )
and the param $scheduled_time is
"@param $scheduled_time int|string a unix timestamp, or a string representing a date"

This could be easier to be addressed in interface for external plugins (I hope)